### PR TITLE
Add merger for retrieve results

### PR DESCRIPTION
**Pull request by XuanYang-cn**
See also: #18893

Signed-off-by: yangxuan <xuan.yang@zilliz.com>

### CODE REVIEW

File: internal/querynode/result.go [hunk 1]
```diff
@@ -35,6 +35,11 @@ import (
 	"github.com/milvus-io/milvus/internal/util/typeutil"
 )
 
+const (
+	unlimited int    = -1
+	maxStr    string = "&*"
```
Comment (jiaoew1991) [inline internal/querynode/result.go:40]: Is there a reference for this value?

Comment (XuanYang-cn) [reply internal/querynode/result.go:40]: No reference, just a special value to indicate it's the max string.

File: internal/querynode/result.go [hunk 2]
```diff
@@ -234,6 +239,63 @@ func encodeSearchResultData(searchResultData *schemapb.SearchResultData, nq int6
...
+	if len(validRetrieveResults) == 0 {
+		return ret, nil
+	}
+
+	if limit != unlimited {
+		loopEnd = limit
+	}
+
+	ret.FieldsData = make([]*schemapb.FieldData, len(validRetrieveResults[0].GetFieldsData()))
+	idSet := make(map[interface{}]struct{})
```
[the remark this answers is not in the document]
Comment (XuanYang-cn) [reply internal/querynode/result.go:272]: The ID here could be int64 or string, so cannot use `typeutil.UniqueSet` whose key is int64.

**Comment by XuanYang-cn** [comment]
rerun ut

**Pull request merged**